### ISSUE TITLE
Magic compose seems not to work

### ISSUE THREAD

**Issue by russianspy1234** [jacopotediosi/GAppsMod#108]
## Overview
[NOTE]: # magic compose does not appear, either in settings or as an option in any message threads. I verified with another app that the flag got set, but it's not doing anything. I also have call recording working so I know the patcher works in general

## Steps to Reproduce
[NOTE]: # 
1.  Start app and grant root
2.  Click magic compose patch
3.  Restart phone
4.  Open messages

## Expected Behavior
[NOTE]: # magic compose bubble appears in message threads

## Actual Behavior
[NOTE]: # no magic compose option appears. 

## System Information
- Device and model: Sony xoeria 1 iii

- ROM and Android version: android 13 61.2.A.0.388 release-keys

- Is the Google app you are trying to tweak (e.g., Phone by Google) installed as system app: no

- Installed Magisk / other SU Manager version: app 26400 / magisk 25200

[NOTE]: # ( Paste below the output of the `adb shell "dumpsys package com.jacopomii.gappsmod | grep version"` command )
- Installed GAppsMod version:    versionCode=400 minSdk=21 targetSdk=33
    versionName=4.00
    signatures=PackageSignatures{bbf04a1 version:2, signatures:[8c539576], past signatures:[]}

[NOTE]: # ( Paste below the output of the `adb shell "dumpsys package REPLACE_WITH_PACKAGENAME | grep version"` command )
- The version of the Google app you are trying to tweak (e.g., Phone by Google):    versionCode=210061100 minSdk=26 targetSdk=34
    versionName=messages.android_20231114_01_RC00.phone.openbeta_dynamic
    signatures=PackageSignatures{bb68e69 version:3, signatures:[2e9b4eb], past signatures:[658fb6d4 flags: 17, 2e9b4eb flags: 17]}
    versionCode=143886063 minSdk=21 targetSdk=33
    versionName=messages.android_20220811_01_RC02.phone_dynamic
    signatures=PackageSignatures{8f0f41c version:0, signatures:[], past signatures:[]}

[NOTE]: # ( Paste below the output of the `adb shell "getprop | grep locale"` command )
- Your device language (locale):[persist.sys.locale]: [en-US]
[ro.product.locale]: [en-US]

[NOTE]: # ( Paste below the output of the `adb shell "getprop | grep iso-country"` command )
- Your location (country of the SIM and country where you are):[gsm.operator.iso-country]: [us,]
[gsm.sim.operator.iso-country]: [us]

**Comment by sawdoctor** [comment]
I have the exact same issue on the pixel fold. Magic compose has never worked for me on android 13 or android 14. I've tried vpns and setting my location to USA in messaging app. The pixel fold should have magic compose natively but only in the USA atm

**Comment by fillwithjoy1** [comment]
Yeah I'm not sure how Google knows the locations to enable these region-specific features, as Google Dialler's automatic call screening is off due to this

**Comment by russianspy1234** [comment]
> Yeah I'm not sure how Google knows the locations to enable these region-specific features, as Google Dialler's automatic call screening is off due to this

I'm in the US so I don't think it's location based

**Comment by fillwithjoy1** [comment]
> > Yeah I'm not sure how Google knows the locations to enable these region-specific features, as Google Dialler's automatic call screening is off due to this
> 
> I'm in the US so I don't think it's location based

Well it has to know the location of where it is, since a lot of the features are geolocation fenced

**Comment by sawdoctor** [comment]
> > > Yeah I'm not sure how Google knows the locations to enable these region-specific features, as Google Dialler's automatic call screening is off due to this
> > 
> > 
> > I'm in the US so I don't think it's location based
> 
> Well it has to know the location of where it is, since a lot of the features are geolocation fenced

I'm in the uk and can enable calll screening, hold for me and direct my call with the mod. Unfortunately even with all the flags enabled I can not enable magic compose so I presumed it was dependent on usa sim card

**Comment by russianspy1234** [comment]
What I mean is, I have a US sim and am in the US.  I was able to enable call screening and call record, but not magic compose. I was also able to make other changes to the Messages app, such as showing the Smart Compose toggle, it's only magic compose that's not working

**Comment by fillwithjoy1** [comment]
> What I mean is, I have a US sim and am in the US. I was able to enable call screening and call record, but not magic compose. I was also able to make other changes to the Messages app, such as showing the Smart Compose toggle, it's only magic compose that's not working

wtf okay, so is it dependent on the SIM one has?
